src/components/rocm/README.md

```diff
@@ -85,3 +85,20 @@ setting the ROCP\_TOOL\_LIB to the PAPI library as follows:
 
 ## Unusual installations
 For the ROCM component to be operational, it must find the dynamic libraries `libhsa-runtime64.so` and `librocprofiler64.so`. These are normally found in the above standard directories. If these libraries are not found (or are not functional) then the component will be listed as "disabled" with a reason explaining the problem. If libraries were not found, then they are not in the expected places.
+
+2. [Device partitioning](#markdown-device-partitioning)
+
+## Device partitioning
+Compute clusters resource managers can partition GPU devices, on compute nodes,
+into subgroups. This means that a job might only see part of the devices on the
+node. How many devices are visible depends on the value of a set of environment
+variables, configured by the resource manager (e.g. HIP\_VISIBLE\_DEVICES,
+ROCR\_VISIBLE\_DEVICES, etc).
+
+In order to detect available devices, the ROCm component relies on the HSA ROCm
+runtime functions (i.e. hsa\_iterate\_agents). The ROCR\_VISIBLE\_DEVICES
+environment variable establishes how many devices will be visible to the ROCm
+runtime. Therefore, by extension, the PAPI ROCm component will only see as many
+devices as allocated by the resource manager through the aforementioned
+environment variable. The component assigns them integer identifiers in the
+range [0, N-1], where N is the number of devices for the partition.
```

Comment:
Add something like:

Therefore, when using the component in an HIP context, the application would need to map the device index given by `hipGetDevice` to this index range and use this index in the event name, e.g., `rocm:::GPUBusy:device=X`. Preferable the UUID of the device should be used for this mapping (see `hipDeviceGetUuid` and `HSA_AMD_AGENT_INFO_UUID`).

Reply:
btw, that could also be a feature that the user uses the UUID in `:device=GPU-DEADBEEFDEADBEEF`. At least `ROCR_VISIBLE_DEVICES` does support this too.

Reply:
CUDA supports this in `CUDA_VISIBLE_DEVICES` too. But I'm not familiar with the CUDA metrics and components

Reply:
@anustuvicl do you have any information for CUDA regarding this?

Reply:
> btw, that could also be a feature that the user uses the UUID in `:device=GPU-DEADBEEFDEADBEEF`. At least `ROCR_VISIBLE_DEVICES` does support this too.

I fear this could confuse users though. Something to keep in mind. We could discuss this further with @jagode and @adanalis too